types/group.js

```diff
@@ -0,0 +1,58 @@
+const GroupMembershipKeys = require('./groupMembershipKeys')
+const Serializable = require('./serializable')
+
+class Group extends Serializable {
```

Comment:
We should adjust object composition here. `Membership` should hold a `Group` instead of the other way around. My thinking here is you do not have a membership without a group, but you can have a group definition without an active membership.

This said, your head is in this much more than mine is, I may be overlooking a reason why this doesn't work.